### PR TITLE
feat(pre-registration): add validation checks for Sponsor fields

### PR DESCRIPTION
**Pull request by vishalvishwanathan-beis**

### CODE REVIEW

File: DVSRegister/Validations/ContactViewModelValidator.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
+﻿using System.Text.RegularExpressions;
+using PhoneNumbers;
+
+namespace DVSRegister.Validations
+{
+	public class ContactViewModelValidator
```
Comment (AiswaryaBEIS) [inline DVSRegister/Validations/ContactViewModelValidator.cs:6]: can we modify the validator to update context directly?   ValidateSponsorViewModel and ValidateContactViewModel can be called from controller. all existing methods in the validator can be changed to private
 public class ContactViewModelValidator
 {

     //Validate Sponsor model call method to validate contact model
     public static void ValidateSponsorViewModel(HttpContext httpContext, SponsorViewModel sponsorViewModel)
     {
         httpContext?.Session.Set("IsSponsorFullNameValid", ValidateFullName(sponsorViewModel?.SponsorFullName));
         httpContext?.Session.Set("IsSponsorEmailValid", ValidateEmail(sponsorViewModel?.SponsorEmail));
        // add reemaining fields
         //call method to validate contact model
         ValidateContactViewModel(httpContext, sponsorViewModel.ContactViewModel);

     }

     /// <summary>
     /// Validates the contact view model only. 
     /// </summary>
     /// <param name="httpContext">The HTTP context.</param>
     /// <param name="contactViewModel">The contact view model.</param>
     public static void ValidateContactViewModel(HttpContext httpContext, ContactViewModel contactViewModel)
      {
             httpContext?.Session.Set("IsFullNameValid", ValidateFullName(contactViewModel.FullName));
             httpContext?.Session.Set("IsEmailValid", ValidateEmail(contactViewModel.Email));
             // add other fields
      }

     /// <summary>
     /// Private method- Validates the full name.
     /// </summary>
     /// <param name="fullName">The full name.</param>
     /// <returns></returns>
     private static bool ValidateFullName(string fullName)
     {
         if (fullName != null && fullName.Length > 0)
         {
             return true;
         }
         else
         {
             return false;
         }
     }
}

File: DVSRegister/Controllers/PreRegistrationController.cs [hunk 2]
```diff
@@ -92,6 +115,32 @@ public IActionResult SaveSponsor(SponsorViewModel sponsorViewModel)
...
+                HttpContext?.Session.SetString("IsFirstVisit", "true");
+            }
+            else
+            {
+                HttpContext?.Session.SetString("IsFirstVisit", "false");
+            }
+
+            if (!(bool)(HttpContext?.Session.Get<bool>("IsSponsorFullNameValid")) || !(bool)(HttpContext?.Session.Get<bool>("IsSponsorEmailValid")) || !(bool)(HttpContext?.Session.Get<bool>("IsSponsorJobTitleValid")) || !(bool)(HttpContext?.Session.Get<bool>("IsSponsorPhoneNumberValid"))
+                || !(bool)(HttpContext?.Session.Get<bool>("IsFullNameValid")) || !(bool)(HttpContext?.Session.Get<bool>("IsEmailValid")) || !(bool)(HttpContext?.Session.Get<bool>("IsJobTitleValid")) || !(bool)(HttpContext?.Session.Get<bool>("IsPhoneNumberValid")))
+            {
```
Comment (AiswaryaBEIS) [inline DVSRegister/Controllers/PreRegistrationController.cs:140]: After modifying the Validator, call method  ContactViewModelValidator(HttpContext,sponsorViewModel)

File: DVSRegister/Controllers/PreRegistrationController.cs [hunk 3]
```diff
@@ -75,13 +78,33 @@ public IActionResult SaveApplicationSponsorSelection(SummaryViewModel viewModel)
         [HttpGet]
         public IActionResult Sponsor()
         {
+            // Retrieve form validation and first visit info from Session
+            ViewBag.IsFirstVisit = HttpContext?.Session.GetString("IsFirstVisit");
+            ViewBag.IsFullNameValid = HttpContext?.Session.GetString("IsFullNameValid");
+            ViewBag.IsEmailValid = HttpContext?.Session.GetString("IsEmailValid");
```
Comment (AiswaryaBEIS) [inline DVSRegister/Controllers/PreRegistrationController.cs:84]: Is it possible to make as private method? private void RetrieveValidationVariables ?

File: DVSRegister/Controllers/PreRegistrationController.cs [hunk 4]
```diff
@@ -101,6 +150,29 @@ public IActionResult SaveContact(ContactViewModel contactViewModel)
...
             summaryViewModel.SponsorViewModel.ContactViewModel = contactViewModel;
             HttpContext?.Session.Set("PreRegistrationSummary", summaryViewModel);
+
+
+
+            HttpContext?.Session.Set("IsFullNameValid", ContactViewModelValidator.ValidateFullName(contactViewModel.FullName));
+            HttpContext?.Session.Set("IsEmailValid", ContactViewModelValidator.ValidateEmail(contactViewModel.Email));
+            HttpContext?.Session.Set("IsJobTitleValid", ContactViewModelValidator.ValidateJobTitle(contactViewModel.JobTitle));
+            HttpContext?.Session.Set("IsPhoneNumberValid", ContactViewModelValidator.ValidatePhoneNumber(contactViewModel.TelephoneNumber));
+
```
[the remark this answers is not in the document]
Comment (vishalvishwanathan-beis) [reply DVSRegister/Controllers/PreRegistrationController.cs:160]: Done

File: DVSRegister/Controllers/PreRegistrationController.cs (hunk 2 above)
Comment (vishalvishwanathan-beis) [reply DVSRegister/Controllers/PreRegistrationController.cs:140]: Done

File: DVSRegister/Validations/ContactViewModelValidator.cs (hunk 1 above)
Comment (vishalvishwanathan-beis) [reply DVSRegister/Validations/ContactViewModelValidator.cs:6]: Done

File: DVSRegister/Controllers/PreRegistrationController.cs (hunk 3 above)
Comment (vishalvishwanathan-beis) [reply DVSRegister/Controllers/PreRegistrationController.cs:84]: Done

**Review by AiswaryaBEIS — APPROVED** [review]

**Review by AiswaryaBEIS — APPROVED** [review]

**Pull request merged**